include/deal.II/numerics/vector_tools_project.templates.h

```diff
@@ -773,19 +775,25 @@ namespace VectorTools
      * Check if we can use the MatrixFree implementation or need
      * to use the matrix based one.
      */
-    template <typename VectorType, int dim>
-    void
+    template <typename VectorType, int dim, int spacedim>
+    typename std::enable_if<
+      !numbers::NumberTraits<typename VectorType::value_type>::is_complex &&
+        dim == spacedim,
+      void>::type
```

Comment:
There is nothing wrong with `std::enable_if`, but it is simpler to use `std::enable_if_t` (because you don't need the `typename` nor the `::type` at the end).